pkgs/development/compilers/ghc/8.4.2-hsc2hs-cross-build.patch

```diff
@@ -0,0 +1,41 @@
+Submodule utils/hsc2hs 9483ad1006..738f3666c8:
```

Comment:
This is a great patch! Can you open a phabricator review of it? Then we can `fetchPatch` it.

Reply:
If it's already an 8.4 branch (or even master) commit we can pull the diff from cgit at haskell.org too. (You can give me hash and I'll do it if you like.)

Reply:
It's on `master` but not `8.4`. It's an update to a submodule, which complicates things. I tried doing `fetchPatch` on `https://git.haskell.org/ghc.git/commitdiff_plain/...`, but the change to the submodule is just shown as `-rev\n+rev`. Had to do `git diff --submodule=diff`, and I don't know of any end point that would have such a diff.

Reply:
https://github.com/haskell/hsc2hs/commit/738f3666c878ee9e79c3d5e819ef8b3460288edf.diff there is a `fetchPath` argument to prefix all paths in the patch so that it works.

Reply:
Fixed, thanks